src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -217,6 +231,30 @@ private void FireOnSendingHeaders()
             }
         }
 
+        private void FireOnResponseCompleted()
+        {
+            List<KeyValuePair<Action<object>, object>> onResponseCompleted = null;
+            lock (_onResponseCompletedSync)
+            {
+                onResponseCompleted = _onResponseCompleted;
+                _onResponseCompleted = null;
+            }
+            if (onResponseCompleted != null)
+            {
+                foreach (var entry in onResponseCompleted)
+                {
+                    try
+                    {
+                        entry.Key.Invoke(entry.Value);
+                    }
+                    catch
+                    {
+                        // Ignore exceptions
```

Comment:
We should log a bug to figure out how to get the ILoggerFactory to the server. Doesn't need to block this PR though.